### PR TITLE
Fix: Navigation links scroll to top on click

### PR DESCRIPTION
**Pull request by Asin-Junior-Honore**
### Summary

Fixes issue #338.

### Description

Updated `ScrollContext.js` to include a scroll-to-top functionality when navigation links are clicked. This ensures that the user is brought to the top of the new page upon navigation.

### How to Test the Changes

1. Clone the repository and switch to the branch where this PR is made:
   ```bash
   git clone https://github.com/TeamShiksha/logoexecutive.git
   cd logoexecutive
   git checkout <branch-name>
   ```

2. Navigate to the client directory and ensure Yarn is installed and dependencies are up to date:
   ```bash
   cd client
   yarn install
   ```

3. Start the application:
   ```bash
   yarn start
   ```

4. Navigate to any page with scrollable content.
5. Click on any navigation link within the application.
6. Verify that the page scrolls to the top upon navigation.

### Context

The issue occurred when users navigated within the application; the page did not reset to the top, causing a poor user experience. This PR addresses this by implementing a scroll-to-top functionality in `ScrollContext.js`.

### Checklist

- [x] I have tested the changes locally and they work as expected.
- [ ] I have added/updated tests that cover the changes. (Update this once tests are added.)
- [ ] I have updated the documentation to reflect the changes.
- [x] This pull request follows the project's coding guidelines.
```

### CODE REVIEW

**Review by aps08 — CHANGES_REQUESTED** [review]
No test file found for the new file added.

**Comment by Asin-Junior-Honore** [comment]
Hey 
@aps08 , I have added a test file for the new ScrollContext. Kindly review at your earliest convenience.

**Comment by aps08** [comment]
Pre-commit is fine. @anandbaraik @samyak269  review this.

File: client/src/App.js [hunk 1]
```diff
@@ -24,47 +25,49 @@ import './App.css';
...
-					element={
-						<ProtectedRoute>
-							<Dashboard />
-						</ProtectedRoute>
-					}
-				/>
-				<Route path='*' element={<NotFound />} />
-			</Routes>
-			<Footer />
+			<ScrollProvider>
```
Comment (anandbaraik) [inline client/src/App.js]: Move <ScrollProvider>  component to `index.js` file. Keep all context provider at one place.

Comment (anandbaraik) [reply client/src/App.js]: > Move component to `index.js` file. Keep all context provider at one place.

@Asin-Junior-Honore

**Comment by anandbaraik** [comment]
@aps08 looked at the PR, tested locally and navigation links scroll to top are working fine.  Test for ScrollContext is passing. 
@Asin-Junior-Honore, I have added a comment, please take a look and change it.
@samyak269 please review it and once it's fine we can merge. CC: @aps08

**Comment by Asin-Junior-Honore** [comment]
hey 
@anandbaraik 
thanks for the review, I would implement the requested changes as soon as possible thanks😊

**Comment by Asin-Junior-Honore** [comment]
hey
@anandbaraik I have made the requested changes. kindly review, thanks😊

**Review by aps08 — APPROVED** [review]

**Comment by aps08** [comment]
@Asin-Junior-Honore Changes are deployed and working properly. And it is working fine. Thank you for your contribution.

**Pull request merged**